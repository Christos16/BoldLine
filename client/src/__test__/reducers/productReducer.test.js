import selectReducer from '../../reducers/productReducer';
import {
  GET_PRODUCTS,
  GET_PRODUCT // Only ones related to the reducer being tested
} from '../../action/types';

//Setting the initial state to 0

describe('select_reducer', () => {
  describe('INITIAL_STATE', () => {
    test('is correct', () => {
      const action = { type: 'dummy_action' };
      const initialState = { products: [], NumberOfProducts: '', product: {} };

      expect(selectReducer(undefined, action)).toEqual(initialState);
    });
  });

  describe('GET_POSTS', () => {
    it('Should return correct state from api', () => {
      const products = [
        {
          size: 1,
          price: 25,
          menCategory: 1,
          womanCategory: 1,
          _id: '12345',
          imagePath: 'http://static.zara',
          title: 'Grey Pullover',
          description: 'Grey Pullover with pockets',
          color: 'Stone',
          quantity: 0,
          department: 'Men'
        },
        {
          size: 2,
          price: 22,
          menCategory: 2,
          womanCategory: 3,
          _id: '12345',
          imagePath: 'http://stfatic.zara',
          title: 'Grey ',
          description: ' pockets',
          color: 'red',
          quantity: 0,
          department: 'Woman'
        }
      ];

      const action = { type: GET_PRODUCTS, payload: products };
      expect(selectReducer(undefined, action)).toMatchSnapshot();
    });

    test('Should render the selected product by id', () => {
      const product = {
        size: 1,
        price: 25,
        menCategory: 1,
        womanCategory: 1,
        _id: '12345',
        imagePath: 'http://static.zara',
        title: 'Grey Pullover',
        description: 'Grey Pullover with pockets',
        color: 'Stone',
        quantity: 0,
        department: 'Men'
      };
      const action = { type: GET_PRODUCT, payload: product };
      expect(selectReducer(undefined, action)).toMatchSnapshot();
    });
  });

  /*
  describe('GET_PRODUCTS', () => {
    test('returns the correct state', () => {
      const action = {
        type: GET_PRODUCTS,
        payload: [
          {
            size: 1,
            price: 25,
            menCategory: 1,
            womanCategory: 1,
            _id: '12345',
            imagePath: 'http://static.zara',
            title: 'Grey Pullover',
            description: 'Grey Pullover with pockets',
            color: 'Stone',
            quantity: 0,
            department: 'Men'
          },
          {
            size: 2,
            price: 22,
            menCategory: 2,
            womanCategory: 3,
            _id: '12345',
            imagePath: 'http://stfatic.zara',
            title: 'Grey ',
            description: ' pockets',
            color: 'red',
            quantity: 0,
            department: 'Woman'
          }
        ]
      };
      const expectedState = {
        products: [
          {
            size: 1,
            price: 25,
            menCategory: 1,
            womanCategory: 1,
            _id: '12345',
            imagePath: 'http://static.zara',
            title: 'Grey Pullover',
            description: 'Grey Pullover with pockets',
            color: 'Stone',
            quantity: 0,
            department: 'Men'
          },
          {
            size: 2,
            price: 22,
            menCategory: 2,
            womanCategory: 3,
            _id: '12345',
            imagePath: 'http://stfatic.zara',
            title: 'Grey ',
            description: ' pockets',
            color: 'red',
            quantity: 0,
            department: 'Woman'
          }
        ]
      };
      expect(selectReducer(undefined, action)).toEqual(expectedState);
    });
   
  });
  */
});
