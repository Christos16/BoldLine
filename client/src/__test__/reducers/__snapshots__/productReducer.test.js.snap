// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`select_reducer GET_POSTS Should render the selected product by id 1`] = `
Object {
  "NumberOfProducts": "",
  "product": Object {
    "_id": "12345",
    "color": "Stone",
    "department": "Men",
    "description": "Grey Pullover with pockets",
    "imagePath": "http://static.zara",
    "menCategory": 1,
    "price": 25,
    "quantity": 0,
    "size": 1,
    "title": "Grey Pullover",
    "womanCategory": 1,
  },
  "products": Array [],
}
`;

exports[`select_reducer GET_POSTS Should return correct state from api 1`] = `
Object {
  "NumberOfProducts": undefined,
  "product": Object {},
  "products": undefined,
}
`;

exports[`select_reducer GET_POSTS Should return default state 1`] = `
Object {
  "NumberOfProducts": undefined,
  "product": Object {},
  "products": undefined,
}
`;
